prov/util/src/util_mr_cache.c

```diff
@@ -316,29 +316,23 @@ util_mr_cache_create(struct ofi_mr_cache *cache, const struct ofi_mr_info *info,
 	return ret;
 }
 
-int ofi_mr_cache_search(struct ofi_mr_cache *cache, const struct fi_mr_attr *attr,
+int ofi_mr_cache_search(struct ofi_mr_cache *cache, const struct ofi_mr_info *info,
```

Comment:
Do we still need to change `ofi_mr_cache_search`?

Reply:
We still need to pass the `IPC_handle` as part of the `mr_entry->info`, so this change is still needed.

Reply:
Why is the ipc_handle needing to be passed down?  What is the core MR cache doing with it?

Reply:
ipc_handle needs to be set in `ofi_mr_entry->info`, so the `ofi_mr_cache_search` can create such mr_entry from the info, and insert it to the cache via `util_mr_cache_create`, right?

The `fi_mr_attr` does not have ipc_handle (and we cannot modify this public API). Is there any alternative we can set ipc_handle into the mr_entry without changing the `ofi_mr_cache_search` API?

Reply:
In our earlier discussion, isn't this code block require the ipc_handle part of the mr_entry in the cache?

```
if (!ipc_handle_valid(entry->ipc_handle))
        update entry->ipc_handle;
```

Reply:
Look at the modified implementation of ofi_mr_cache_search()/ find()/ reg().  They don't touch the ipc_handle or access any data that they already didn't have access to.  The ipc_handle is only accessed by the wrapper call ofi_ipc_cache_search().  I don't think the core mr cache needs to change.  We just need to add the ipc fields to ofi_mr_entry, so the ipc cache has somewhere to store the data.

Reply:
But currently `ofi_mr_cache_search` construct the `ofi_mr_info` inside the function, by copying the iov from `fi_mr_attr` in the input? If you do not change the api, the constructed `ofi_mr_info` inside the function will not have ipc_handle, right?

Reply:
Yes, if possible.  That isolates IPC handling to the ipc level cache.